x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmSettingsTests.java

```diff
@@ -0,0 +1,374 @@
+import org.junit.Before;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class JwtRealmSettingsTests extends ESTestCase {
+
+    private static final String REALM_NAME = "jwt1-realm";
+    private ThreadContext threadContext;
+
+    @Before
+    public void setupEnv() {
+        Settings globalSettings = Settings.builder().put("path.home", createTempDir()).build();
+        threadContext = new ThreadContext(globalSettings);
+    }
+
+    public void testAllSettings() {
+        final RealmConfig realmConfig = getMockWorkingSettings();
+        JwtRealmSettingsTests.validateSettings(realmConfig, JwtRealmSettings.getSettings());
+    }
+
+    public void testAllowedIssuer() {
+        final Setting.AffixSetting<String> setting = JwtRealmSettings.ALLOWED_ISSUER;
+        final String settingKey = RealmSettings.getFullSettingKey(REALM_NAME, setting);
+        for (final String rejectedValue : new String[] { null, "" }) {
+            final Exception exception = expectThrows(IllegalArgumentException.class, () -> {
+                final Settings settings = Settings.builder().put(settingKey, rejectedValue).build();
+                final String actualValue = this.buildConfig(settings).getSetting(setting);
+                fail("No exception. Expected one for " + settingKey + "=" + rejectedValue + ". Got " + actualValue + ".");
+            });
+            assertThat(exception.getMessage(), equalTo("Invalid null or empty value for [" + settingKey + "]."));
+        }
+        for (final String acceptedValue : new String[] { "http://localhost/iss1", "issuer1", "i" }) {
+            final Settings settings = Settings.builder().put(settingKey, acceptedValue).build();
+            final String actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(acceptedValue));
+        }
+    }
+
+    public void testAllowedSignatureAlgorithms() {
+        Setting.AffixSetting<List<String>> setting = JwtRealmSettings.ALLOWED_SIGNATURE_ALGORITHMS;
+        final String settingKey = RealmSettings.getFullSettingKey(REALM_NAME, setting);
+        final String allCsv = "HS256, HS384, HS512, RS256, RS384, RS512, ES256, ES384, ES512, PS256, PS384, PS512";
+        for (final String rejectedValue : new String[] { "unknown", "HS256,unknown" }) {
+            final Exception exception = expectThrows(IllegalArgumentException.class, () -> {
+                final Settings settings = Settings.builder().put(settingKey, rejectedValue).build();
+                final List<String> actualValue = this.buildConfig(settings).getSetting(setting);
+                fail("No exception. Expected one for " + settingKey + "=" + rejectedValue + ". Got " + actualValue + ".");
+            });
+            assertThat(
+                exception.getMessage(),
+                equalTo("Invalid value [unknown] for [" + settingKey + "]." + " Allowed values are [" + allCsv + "]}].")
```

Comment:
you can pass ALLOWED_SIGNATURE_ALGORITHMS directly and you don't need the extra brackts around it

Reply:
I agree. I made the change.